bin/test-isolated

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env node
+
+// Basic isolated tests runner
+// Ensures each test file is run in distinct process and does not interfere with other test runs.
+// To be used to confirm test files do not introduce and work by chance of side effects
+// Temporary solution until we migrate to runner which provides that (reliably) on its own
+
+'use strict';
+
+process.on('unhandledRejection', err => {
+  throw err;
+});
+
+const globby = require('globby');
+const spawn = require('child-process-ext/spawn');
+const chalk = require('chalk');
+
+const patterns = process.argv.length <= 2 ? ['**/*.test.js'] : process.argv.slice(2);
+patterns.push('!node_modules/**');
+
+const resolveGitStatus = () =>
+  spawn('git', ['status', '--porcelain']).then(
+    ({ stdoutBuffer }) => String(stdoutBuffer),
+    error => {
+      process.stdout.write(error.stdoutBuffer);
+      process.stderr.write(error.stderrBuffer);
+      throw error;
+    }
+  );
+
+const initialGitStatusDeferred = resolveGitStatus();
+
+const run = path => {
+  const onFinally = () =>
+    Promise.all([initialGitStatusDeferred, resolveGitStatus()]).then(
```

Comment:
What's the difference between the two Promises here? I understand what this code should do but aren't they technically the same?

Reply:
First one exposes `git status` as it was _before_ test was run, and second one resolves one _after_ test was run.

If there's a difference between both, it means some temporary files weren't cleaned up